### PR TITLE
refactoring source code

### PR DESCRIPTION
**Pull request by skaghzz**
We did refactoring source code to lower Cyclomatic complexity and Nested Block Depth and to enhance cohesion and readability.

### CODE REVIEW

**Comment by tonysparks** [comment]
Thank you for submitting this merge request.  There are a couple of things I recommend fixing prior to merging:

* Replace all tabs with 3 spaces (i.e., no tabs); this allows for consistent formatting
* Method names should be camelCase(); this is consistent with Java standards
* Unless explicitly required to be public, don't create getters/setters for private data members;  this just adds cruft to the code, doesn't enhance readability
* Variables should be defined as close to where they are used; this increases readability and in some cases is slightly more optimal

Much appreciated.

**Comment by skaghzz** [comment]
Okad. I will Comment a message after doing that

**Comment by skaghzz** [comment]
We did all up all of you requirements.

**Comment by skaghzz** [comment]
Hi, Tony.
are there any matter to merge our request?

**Comment by tonysparks** [comment]
Hello skaghzz,

Sorry I haven't had time to fully check the merge request just yet.  I did notice quite a bit of formatting issues still.  Another concern of mine is changing some of the branching logic -- which will require testing and I'm not sure what advantage it provides.

I'll be sure to check it out more this weekend and provide more detailed feedback.

Much appreciated!

**Comment by skaghzz** [comment]
hello tony,
I'm still waiting for your feedback.

**Comment by skaghzz** [comment]
Thanks to your feedback.
I read your comments and resolve the problems.

**Comment by tonysparks** [comment]
Thank you skaghzz, I should have some time to review some more tonight and tomorrow.

**Comment by skaghzz** [comment]
you're welcome.
If there is more feedback, plz comment.

File: src/seventh/ai/basic/actions/atom/body/SwitchWeaponAction.java [hunk 1]
```diff
@@ -49,27 +49,15 @@ public void update(Brain brain, TimeStep timeStep) {
...
-        }
-        
-        if(isDone) {
-            getActionResult().setSuccess();
-        }
-        else {
-            getActionResult().setFailure();
-        }
-        
+       if (currentItem ==null||currentItem.getType().equals(weapon)){
```
Comment (tonysparks) [inline src/seventh/ai/basic/actions/atom/body/SwitchWeaponAction.java]: Formatting here is a bit off

File: src/seventh/game/Players.java [hunk 2]
```diff
@@ -1,256 +1,277 @@
...
+    /* (non-Javadoc)
+     * @see seventh.game.PlayerInfos#getPrevAlivePlayerFrom(seventh.game.Player)
+     */
+    
+    
+    @Override
+    public Player getPrevAlivePlayerFrom(Player oldPlayer) {
+        if(oldPlayer == null ) return getRandomAlivePlayer();
+        int PlayerSize = players.length;
+        int nextPlayerIndex = SetNextPlayerIndex(oldPlayer, PlayerSize);
```
Comment (tonysparks) [inline src/seventh/game/Players.java]: method names should be camelCase

File: src/seventh/game/Players.java [hunk 3]
```diff
@@ -1,256 +1,277 @@
...
+    	return player != null && player.isAlive() && player != oldPlayer;
+    }
+    
+    
+    public int SetNextPlayerIndex(Player oldPlayer, int PlayerSize){
+    	int nextPlayerIndex = (oldPlayer.getId() -1) % PlayerSize;
+    	return isUnderPlayerIndex(nextPlayerIndex, PlayerSize);
+    }
+    
+    public int changePlayerIndex(int nextPlayerIndex, int PlayerSize){
```
Comment (tonysparks) [inline src/seventh/game/Players.java]: variable names should be camelCase.  Also this method probably should be private

**Comment by tonysparks** [comment]
It still seems like there are a lot of formatting issues.

**Comment by skaghzz** [comment]
I resolve formatting issue and standard conventions problems

**Comment by skaghzz** [comment]
Hello, Tony.
We have become contributors for our spring semester term project making test case, refactoring and design pattern. but spring semester will be ended soon after 1 weeks. So, there is no time to wait merging our request. If you give more feedback, we correspond immediately.
thanks.

**Comment by tonysparks** [comment]
Hi Skaghzz,

I really appreciate your efforts and contributions.  I will try my hardest to approve this in a timely manner.  I'm still seeing a lot of formatting issues (tabs, alignment) in the source code.  Please take a closer look at all the files, once that is complete I can give a more detailed review of the code.

Much appreciated,

Tony

**Pull request closed without merging**